### DISCUSSION TITLE
Deep linking not working with Vite/React deploy

### DISCUSSION THREAD

**Discussion by nareshbhatia** [vercel/community#149] [Help] [answered]
Hi folks, I have deployed a React app using Vite to Vercel. Everything works fine, except that when I deep link to any page, I get a 404. Is there a Vercel setting that I am missing? Here's the deployed app: https://nested-routes-react-router.vercel.app/

**Comment by nareshbhatia** [comment, marked as answer]
Found the answer. Added the following configuration in vercel.json:

```
{
  "rewrites": [{ "source": "/(.*)", "destination": "/index.html" }]
}
```

**Reply by yamiteru** [reply]
Where do you put this file?

**Reply by nareshbhatia** [reply]
In the root folder. See here: https://github.com/nareshbhatia/nested-routes/tree/main/nested-routes-react-router

**Reply by styfle** [reply]
Have you tried configuring Vite to use "Multi-Page App" mode?
Its recommended for production builds.
See https://vitejs.dev/guide/build.html#multi-page-app

**Reply by Plunzi** [reply]
Very usefull, thank you!

**Reply by NavedKhan11** [reply]
@nareshbhatia 
Thanks a lot it is working fine now.
You saved my so much hours.

**Reply by elioenays** [reply]
thank you

**Comment by ashutosh887** [comment]
Thanks @nareshbhatia
